src/safeposix/syscalls/fs_calls.rs

```diff
@@ -988,118 +1036,122 @@ impl Cage {
             &writer
         };
 
-        if let Some(wrappedfd) = fdtable.get(&fd) {
-            let filedesc_enum = wrappedfd.read().unwrap();
-            let mut mutmetadata = FS_METADATA.write().unwrap();
+        let filedesc_enum = fdtable.get(&fd).unwrap().read().unwrap();
+        let mut mutmetadata = FS_METADATA.write().unwrap();
 
-            //Decide how to proceed depending on the fd type.
-            //First we check in the file descriptor to handle sockets, streams, and pipes,
-            //and if it is a normal file descriptor we decrement the refcount to reflect
-            //one less reference to the file.
-            match &*filedesc_enum {
-                //if we are a socket, we dont change disk metadata
-                Stream(_) => {return 0;},
-                Socket(_) => {
-                    Self::_cleanup_socket(self, &fd, false);
-                 },
-                Pipe(pipe_filedesc_obj) => {
-                    let pipenumber = pipe_filedesc_obj.pipe;
-                    let read_references = Self::_lookup_refs_by_pipe_end(self, pipenumber, O_RDONLY);
-                    let write_references = Self::_lookup_refs_by_pipe_end(self, pipenumber, O_WRONLY);
+        //Decide how to proceed depending on the fd type.
+        //First we check in the file descriptor to handle sockets, streams, and pipes,
+        //and if it is a normal file descriptor we decrement the refcount to reflect
+        //one less reference to the file.
+        match &*filedesc_enum {
+            //if we are a socket, we dont change disk metadata
+            Stream(_) => {return 0;},
```

Comment:
Evn if you don't chang disk metadata, don't return here because you still need to remove from fdtable